app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt

```diff
@@ -317,6 +330,37 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
         return true
     }
 
+    fun processDeepLink(deepLinkInfo: DeepLinkInfo) {
+        val username = deepLinkInfo.roomName
+        username.ifNotNullNotEmpty {
+            val localRooms = viewModel.getChatRoomOfUsernameDB(username!!)
+            val filteredLocalRooms = localRooms.filter { itemHolder -> itemHolder.data is RoomUiModel && (itemHolder.data as RoomUiModel).username == username }
+
+            if (filteredLocalRooms.isNotEmpty()) {
+                presenter.loadChatRoom(filteredLocalRooms.first().data as RoomUiModel)
+            } else {
+                //check from spotlight when connected
+                val statusLiveData = viewModel.getStatus()
+                statusLiveData.observe(viewLifecycleOwner, object: Observer<State>{
+                    override fun onChanged(status: State?) {
+                        if (status is State.Connected) {
+                            val rooms = viewModel.getChatRoomOfUsernameSpotlight(username)
+                            val filteredRooms = rooms?.filter { itemHolder -> itemHolder.data is RoomUiModel && (itemHolder.data as RoomUiModel).username == username }
+
+                            filteredRooms?.let {
+                                if (filteredRooms.isNotEmpty()) {
+                                    presenter.loadChatRoom(filteredRooms.first().data as RoomUiModel)
+                                }
+                            }
+
+                            statusLiveData.removeObserver(this)
+                        }
```

Comment:
For all the `}`, '})' at the end of this function here (and on the fragment) I believe we can improve this code.

Reply:
I broke this up a little bit, fwiw